File: ARCHISOFT_topoalign/Pick Filters/LinePickFilter.vb
Public Class LinePickFilter
    Implements ISelectionFilter

    Public Function AllowElement(elem As Element) As Boolean Implements ISelectionFilter.AllowElement
        ' Convert the element to a ModelLine
        Dim line As ModelLine = TryCast(elem, ModelLine)
        Dim curve As ModelCurve = TryCast(elem, ModelCurve)
        ' line is null if the element is not a model line
        If line Is Nothing And curve Is Nothing Then
            Return False
        End If
        ' return true if the line is a model line
        Return True
    End Function

    Public Function AllowReference(reference As Reference, position As XYZ) As Boolean Implements ISelectionFilter.AllowReference
        Return False
    End Function
End Class

File: ARCHISOFT_topoalign/Pick Filters/LinePickFilter.cs
using Autodesk.Revit.DB;
using Autodesk.Revit.UI.Selection;

namespace ARCHISOFT_topoalign
{
    public class LinePickFilter : ISelectionFilter
    {
        public bool AllowElement(Element elem)
        {
            // Convert the element to a ModelLine
            ModelLine line = elem as ModelLine;
            ModelCurve curve = elem as ModelCurve;
            // line is null if the element is not a model line
            if (line is null & curve is null)
            {
                return false;
            }
            // return true if the line is a model line
            return true;
        }

        public bool AllowReference(Reference reference, XYZ position)
        {
            return false;
        }
    }
}